Frontend/src/Components/Checkout/PaymentPage.jsx

```diff
@@ -0,0 +1,149 @@
+import React, { useEffect, useState } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+import { useStripe, useElements, CardElement } from "@stripe/react-stripe-js";
+import { loadStripe } from "@stripe/stripe-js";
+import AppHeader from "../Headers/AppHeader";
+import "./PaymentPage.css";
+
+const API_KEY = import.meta.env.VITE_BACKEND_ADDRESS;
+
+const PaymentPage = () => {
+  const { id } = useParams();
+  const stripe = useStripe();
+  const elements = useElements();
+  const navigate = useNavigate();
+  const [clientSecret, setClientSecret] = useState("");
+  const [listing, setListing] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [success, setSuccess] = useState(false);
+
+  useEffect(() => {
```

Comment:
so for my previous comment, i'm seeing that we're fetching the listing a lot so i'm thinking making this a child route would speed up performance since we don't need to access the API and can directly access listing from useOutletContext